### DISCUSSION TITLE
Remove DNS flattening requirement for apex domain

### DISCUSSION THREAD

**Discussion by matthewstick** [craftcms/cloud#13] [Feature Requests & Enhancements]
There is a note "Your DNS provider must support CNAME flattening (sometimes referred to as ANAME or ALIAS records) to use a CNAME with an apex domain"

Route53 in AWS does not support DNS flattening.  They do have an "alias hack" that lets you cname the apex to AWS owned resources, but not external resources.

Route53 is a really common DNS so it would be nice if we could figure out how to host this.  We can get the IP address from you, but this may change and is not ideal.

**Comment by timkelty** [comment]
I agree it would be nice, but we're a bit bound by the higher powers, here.
CNAME'ing the apex goes against some foundations of DNS, which is why "hacks" like ALIAS exist.

In a happy internet, everybody uses www. for web traffic, but that reality clearly didn't happen.

The closest we could get to supporting this, I think, would be to offer a site.com -> www.site.com redirect, but if you really want your canonical to be at the apex, you're going to be out of luck.

Alas, we'll keep our eyes open for any clever workarounds. We fully agree this is annoying.
